File: python/4/35.py
from functions4 import *

long_nouns = []

for sentence in morphological():
    nouns = []

    for word in sentence:
        if word["pos"] == "名詞":
            nouns.append(word["surface"])
        else:
            if len(nouns) > 1:
                long_nouns.append(nouns)
            nouns = []

for nouns in long_nouns:
    print("".join(nouns))

File: ruby/4/35.rb
require './util'

long_nouns = []

morphologies.each do |sentence|
  nouns = []
  sentence.each do |pos:, surface:, **|
    if pos == "名詞"
      nouns << surface
    else
      long_nouns << nouns if nouns.length > 1
      nouns = []
    end
  end
end

puts long_nouns.map(&:join)
